### PR TITLE
Convert Game to SGF (Issue#127)

### PR DESCRIPTION
**Pull request by TrevorPeters**

### CODE REVIEW

**Comment by TrevorPeters** [comment]
Interesting. the test passes when I run it in Vs code, but fails when I run it in the command line. will investigate further.

Update: the test passes and fails randomly. Vs code cached the first successful result leading me to believe that it was consistently passing. I need to change the testing method

Update 2: testing issue resolved :D

File: core/sgf/parser.go [hunk 1]
```diff
@@ -395,3 +395,38 @@ func postProcessProperties(n *game.Node, prop string, propData []string) error {
 
 	return nil
 }
+
+// Serialize converts a Game into SGF format.
+// Calls serializeHelper.
```
Comment (artasparks) [inline core/sgf/parser.go:400]: nit: You don't need to mentioned the internals (serializeHelper) in the public docs.

File: core/sgf/parser.go [hunk 2]
```diff
@@ -395,3 +395,38 @@ func postProcessProperties(n *game.Node, prop string, propData []string) error {
...
+func Serialize(g *game.Game) string {
+	return fmt.Sprintf("(%s)", serializeHelper(g.Root))
+}
+
+// serializHelper is a recursive DFS searching all
+// descendant nodes of n.
+func serializeHelper(n *game.Node) string {
```
Comment (artasparks) [inline core/sgf/parser.go:407]: This is very slick, nice.

**Review by artasparks — APPROVED** [review]
Nice!

**Pull request merged**